### DISCUSSION TITLE
Deleted by the user Autoadded Product

### DISCUSSION THREAD

**Discussion by elektro-hub** [EvgeniyMukhamedjanov/liquid-ajax-cart#85] [Q&A] [answered]
Hi ,
I was just testing and found out if the product was added with the mutation option then its not deleted anymore even if someone try to delete it then its again there after few seconds. Is there a way to track this and eventually not to add the product again if its deleted by the user?

Cheers,
Petar

**Comment by EvgeniyMukhamedjanov** [comment, marked as answer]
Hi @elektro-hub 
Yes, you can subscribe to the [`liquid-ajax-cart:request-end`](https://liquid-ajax-cart.js.org/v2/event-request-end/) event, check if the autoadded item has been just removed removed using the `cart` property of the event and add cookie or a localStorage value to the user that will indicate that they don't wanna have the autoadded item:

```javascript
// Subscribe to 
document.addEventListener("liquid-ajax-cart:request-end", function(event) {
  const {cart, requestState} = event.detail;

  // Variant Id of the autoadded product
  const variantId = 40934235668668 

  // Has autoadded product been just removed
  const isAutoAddedItemRemoved = cart.items_removed?.find(item => item.variant_id === variantId);
  if (!isAutoAddedItemRemoved) {
    return false;
  }

  // If the product has been removed by the user action rather by a mutation
  const removedByUser = requestState.info.initiator !== "mutation";
  if (!removedByUser) {
    return false;
  }

  // Save data to localStorage that the user removed the product
  localStorage.setItem("autoadded_removed", "true");
});

// Also update your mutation function:
function myMutation() {
  // Right in the beginning stop the mutation function if the user has localStorage value
  if (localStorage.getItem("autoadded_removed") === "true") {
    return;
  }
  ...
```

**Reply by elektro-hub** [reply]
Thanks, 
this works ! :)
